platform/core/build_for_android.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+set -e    # Exit if any command fails
+
+# ----------------------- Command options ------------------------ #
+android_version=24
+ndk_version=r22b
+build_type=debug
+release_flag=""
+
+while [ $# -gt 0 ]; do
+
+   if [[ $1 == *"--"* ]]; then
+        param="${1/--/}"
+        declare $param="$2"
+   fi
+
+  shift
+done
+
+if [[ $build_type == release ]]; then
+    release_flag="--release"
+fi
+
+# ----------------------- NDK setup ------------------------ #
+google_repo=https://dl.google.com/android/repository
+ndk_lib=android-ndk-${ndk_version}
+android_export=export/android
+ndk_dir=${android_export}/NDK
+# Override NDK_HOME to point to downloaded version.
+# Will be used by `cargo ndk` to generate the libraries
+export NDK_HOME=${ndk_dir}/${ndk_lib}
+
+if [ ! -d $NDK_HOME ]; then
+	mkdir -p ${android_export}
+
+	echo "*** Installing NDK $ndk_version"
+	ndk_download_link=$google_repo/$ndk_lib
+	os_name=""
+	case "$OSTYPE" in
+	    darwin*)
+	        os_name=darwin
+	    ;;
+	    linux*)
+	        os_name=linux
+	    ;;
+	    msys*)
+	        os_name=windows
+	    ;;
+	    cygwin*)
+	        os_name=windows
+	    ;;
+	    *)
+	        echo "unsupported OS: $OSTYPE"
+	        exit 1
+	    ;;
+	esac
+
+	ndk_download_link=$google_repo/$ndk_lib-$os_name-x86_64.zip 
```

Comment:
Would that actually work on M1 macbooks?

I'm pretty sceptical about fetching the NDK as part of the build - it's like 1GB... But I think I wanna leave it like that at first.

Reply:
Don't M1 machines emulate that architecture? 
Would be nice to have a native version, anyway...

Reply:
As far as I know, Apple M1 support is available starting with NDK version r23. But with rust we still have to use NDK r22. There are some workaround to make Rust properly work with NDK `>=r23`, but I think it is better to use stable `r22` version. Later versions of NDK, `>=r23` do not require specifying the architecture, so when NDK version will be updated, we will remove the architecture parameter.

Yes, it works on M1 MacBooks through Rosetta, and it is the accepted way to run NDK `<r23` version.